### DISCUSSION TITLE
aws ec2 describe instances output is not getting filtered

### DISCUSSION THREAD

**Discussion by lesh3000** [aws/aws-cli#7453] [Q&A] [answered]
### Describe the bug

I am using bash and calling:
`aws ec2 describe-instances --region eu-central-1 --filters "Name=tag-key,Values=Name" --query "Reservations[*].Instances[*].Placement[*]"`

However te result is empy.
If I do `aws ec2 describe-instances --region eu-central-1 --filters "Name=tag-key,Values=Name" --query "Reservations[*].Instances[*].PrivateDnsName")` it works as it should

### Expected Behavior

List of placement objects

### Current Behavior

Non

### Reproduction Steps

aws ec2 describe-instances --region eu-central-1 --filters "Name=tag-key,Values=Name" --query "Reservations[*].Instances[*].Placement[*]"

### CLI version used

aws-cli/1.18.147 Python/2.7.18 Linux/5.10.144-127.601.amzn2.x86_64 botocore/1.18.6

### Environment details (OS name and version, etc.)

run on EC2 instance via console

**Comment by tim-finnigan** [comment]
Hi @lesh3000 thanks for reaching out. Here is the `ec2 describe-instances` documentation for reference: https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-instances.html

I think the snippet you're looking for is something like: 
```
aws ec2 describe-instances --region eu-central-1 --filter Name=tag:Name,Values=<your_value_here> --query "Reservations[].Instances[].Placement"
```
Does that accomplish what you're trying to do? Please let us know.

**Comment by lesh3000** [comment]
Hi,
Thank you for your prompt reply.
It seems working if I call it this way: `--query "Reservations[].Instances[].Placement[].AvailibilityZone[]" `
I need to get AZ only.
The docs show mostly [*] usage in the end of the json property, where can I read about \[\] vs \[\*\] vs none in the end?
Appreciate your help

**Comment by tim-finnigan** [comment, marked as answer]
Thanks @lesh3000 for following up. You can read more about client-side filtering with the AWS CLI here: https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-filter.html#cli-usage-filter-client-side. 

Also since this was a guidance-related question, I'm going to convert it to a GitHub discussion per our issue triage process. Please let us know if you had any additional questions or feedback.
